converter/internal/prometheusconvert/prometheusconvert.go

```diff
@@ -86,105 +84,53 @@ func AppendAll(f *builder.File, promConfig *prom_config.Config, labelPrefix stri
 		appendPrometheusScrape(pb, scrapeConfig, scrapeForwardTo, scrapeTargets, label)
 	}
 
-	prepareFileBlocks(f, pb)
-	return diags
+	pb.appendToFile(f)
+
+	return validate(promConfig)
 }
 
 // appendServiceDiscoveryConfigs will loop through the service discovery
 // configs and append them to the file. This returns the scrape targets
 // and discovery targets as a result.
-func appendServiceDiscoveryConfigs(pb *prometheusBlocks, serviceDiscoveryConfig prom_discover.Configs, label string) ([]discovery.Target, diag.Diagnostics) {
-	var diags diag.Diagnostics
+func appendServiceDiscoveryConfigs(pb *prometheusBlocks, serviceDiscoveryConfig prom_discover.Configs, label string) []discovery.Target {
```

Comment:
no longer determine diags as blocks are appended. this is managed by a call to validate afterwards.